### PR TITLE
add unregistered-user dialog

### PR DESCRIPTION
**Pull request by tomivm**
the messages are prototypes without translate.

### CODE REVIEW

File: src/components/Board/BoardShare/BoardShare.component.js [hunk 1]
```diff
@@ -26,6 +27,12 @@ import messages from './BoardShare.messages';
 
 import './BoardShare.css';
 
+function OpenLogWd(){
+  window.open(window.location.origin + '/login-signup','',"width=500,height=800");
```
Comment (martinbedouret) [inline src/components/Board/BoardShare/BoardShare.component.js]: You have to use react router to move to a different location inside the application, see an example here:
https://github.com/cboard-org/cboard/blob/51035b0a503a9f7c3903e13fcd05f2895e89f57c/src/components/Board/Navbar/Navbar.js#L172

React router: https://reactrouter.com/web/guides/quick-start

File: src/components/Board/BoardShare/BoardShare.component.js [hunk 2]
```diff
@@ -26,6 +27,12 @@ import messages from './BoardShare.messages';
 
 import './BoardShare.css';
 
+function OpenLogWd(){
```
Comment (martinbedouret) [inline src/components/Board/BoardShare/BoardShare.component.js]: function names should always start with lower case. See coding convention: https://www.robinwieruch.de/javascript-naming-conventions

File: src/components/Board/BoardShare/BoardShare.component.js [hunk 3]
```diff
@@ -67,28 +74,36 @@ const BoardShare = ({
...
+            isOwnBoard  
+              ? <Button
+              color="primary"
+              variant={isPublic ? "outlined" : "contained"}
+              className="ShareDialog__ToggleStatusButton"
+              onClick={publishBoard}
+              >
+                {!isPublic? <FormattedMessage {...messages.publishBoard } /> : <FormattedMessage {...messages.unpublishBoard } />}
+              </Button>
+              : <><Alert severity="warning">warning — you need to be registered to share a board</Alert>
```
Comment (martinbedouret) [inline src/components/Board/BoardShare/BoardShare.component.js]: you have to use a string from the messages file in order to support translations

File: src/components/Board/BoardShare/BoardShare.component.js [hunk 4]
```diff
@@ -67,28 +74,36 @@ const BoardShare = ({
...
+                <Button
+                  color="primary"
+                  variant="contained"
+                  onClick={OpenLogWd}
+                >LOGIN or SING UP</Button></>
```
Comment (martinbedouret) [inline src/components/Board/BoardShare/BoardShare.component.js]: you have to use a string from the messages file in order to support translations

File: src/components/Board/BoardShare/BoardShare.component.js [hunk 5]
```diff
@@ -67,28 +74,36 @@ const BoardShare = ({
...
           }
           </div>
         </div>
-
+        {isSharable = isPublic && isOwnBoard}
```
Comment (martinbedouret) [inline src/components/Board/BoardShare/BoardShare.component.js]: never do assignments in a jsx node!!!

**Review by martinbedouret — CHANGES_REQUESTED** [review]

File: src/components/Board/BoardShare/BoardShare.component.js (hunk 3 above)
Comment (tomivm) [reply src/components/Board/BoardShare/BoardShare.component.js]: yes, i was thinking on doing that before merging the brach.

**Pull request closed without merging**